### PR TITLE
Add support for wifi and bluetooth

### PR DESCRIPTION
**Pull request by vj-kumar**
Testing wifi:
`
sudo nmcli dev wifi connect <ssid name> password <password>`

Testing bluetooth:
```
bluetoothctl
# scan on
# pair <mac address>
```

### CODE REVIEW

**Comment by chombourger** [comment]
SPDX headers are missing
Find out which with:
pip3 install --user reuse
reuse lint

File: meta-isar/recipes-bsp/firmware-nanopi-r1/files/postinst [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#!/bin/sh
+
+set -e
+
+NANOPI_NVRAM_CONFIG="/lib/firmware/brcm/brcmfmac43430-sdio.friendlyarm,nanopi-r1.txt"
+AP6212_NVRAM_CONFIG="/lib/firmware/brcm/brcmfmac43430-sdio.AP6212.txt"
+if [ ! -f "${NANOPI_NVRAM_CONFIG}" ] && [ -f "${AP6212_NVRAM_CONFIG}" ]; then
+    ln -sf "${AP6212_NVRAM_CONFIG}" "${NANOPI_NVRAM_CONFIG}"
+    mkdir -p /usr/share/firmware-nanopi-r1/
+    touch /usr/share/firmware-nanopi-r1/symlink
```
Comment (chombourger) [inline meta-isar/recipes-bsp/firmware-nanopi-r1/files/postinst:10]: Not sure why we need this dummy symlink file?

File: meta-isar/recipes-bsp/firmware-nanopi-r1/files/prerm [hunk 2]
```diff
@@ -0,0 +1,10 @@
+#!/bin/sh
+
+set -e
+
+NANOPI_NVRAM_CONFIG="/lib/firmware/brcm/brcmfmac43430-sdio.friendlyarm,nanopi-r1.txt"
+
+if [ -f /usr/share/firmware-nanopi-r1/symlink ]; then
```
Comment (chombourger) [inline meta-isar/recipes-bsp/firmware-nanopi-r1/files/prerm:7]: Could be replaced with

if [ -h <mysdio.txt> ]; then
    rm -f <mysdio.txt>
fi

File: meta-isar/conf/machine/nanopi-r1.conf [hunk 3]
```diff
@@ -14,3 +14,6 @@ WKS_FILE ?= "nanopi-r1.wks.in"
 
 IMAGE_INSTALL += "u-boot-script"
 IMAGER_INSTALL += "u-boot-nanopi-r1"
+
+IMAGE_PREINSTALL += " wireless-regdb wireless-tools bluetooth"
```
Comment (chombourger) [inline meta-isar/conf/machine/nanopi-r1.conf:18]: I would be tempted to add all generic wireless packages to mtda-image.inc

File: meta-isar/recipes-bsp/firmware-nanopi-r1/files/postinst (hunk 1 above)
Comment (vj-kumar) [reply meta-isar/recipes-bsp/firmware-nanopi-r1/files/postinst:10]: To know if this package has installed the symlink or whether its from upstream.

File: meta-isar/conf/machine/nanopi-r1.conf (hunk 3 above)
Comment (vj-kumar) [reply meta-isar/conf/machine/nanopi-r1.conf:18]: Sure.

File: meta-isar/recipes-bsp/firmware-nanopi-r1/files/postinst (hunk 1 above)
Comment (vj-kumar) [reply meta-isar/recipes-bsp/firmware-nanopi-r1/files/postinst:10]: As of date there is no such symlink upstream though.

Comment (vj-kumar) [reply meta-isar/recipes-bsp/firmware-nanopi-r1/files/postinst:10]: I think there is a better way to do this. Let me try that and update the PR

Comment (vj-kumar) [reply meta-isar/recipes-bsp/firmware-nanopi-r1/files/postinst:10]: We can use dpkg-query to make sure that symlink doesnot belong to any other package.

File: meta-isar/recipes-bsp/firmware-nanopi-r1/firmware-nanopi-r1.bb [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+
+DESCRIPTION = "Wifi firmware symlink for NanoPi R1"
+MAINTAINER = "mtda-users <mtda-users@googlegroups.com>"
+DEBIAN_DEPENDS = "firmware-brcm80211"
+
+SRC_URI = " file://postinst \
+        file://prerm \
+        file://rules"
+
+inherit dpkg-raw
```
Comment (chombourger) [inline meta-isar/recipes-bsp/firmware-nanopi-r1/firmware-nanopi-r1.bb:13]: A possibly easier way is to ship the symbolic link and let apt report an error in the unlikely event where two packages are shipping `brcmfmac43430-sdio.friendlyarm,nanopi-r1.txt`.

Here's how I reworked your recipe:
```
# Copyright (C) 2021 Siemens Digital Industries Software
# SPDX-License-Identifier: MIT

inherit dpkg-raw

DESCRIPTION = "Wi-Fi firmware suport for NanoPi R1"
MAINTAINER = "mtda-users <mtda-users@googlegroups.com>"
DEBIAN_DEPENDS = "firmware-brcm80211"

do_install() {
    targetdir=${D}/lib/firmware/brcm
    targetlnk=${targetdir}/brcmfmac43430-sdio.friendlyarm,nanopi-r1.txt
    install -m 0755 -d ${targetdir}
    ln -s brcmfmac43430-sdio.AP6212.txt ${targetlnk}
}
```
We may then remove `postinst`, `prerm` and `rules`

Comment (chombourger) [reply meta-isar/recipes-bsp/firmware-nanopi-r1/firmware-nanopi-r1.bb:13]:
```
root@mtda:/lib/firmware/brcm# ls -l
total 11668
...
-rw-r--r-- 1 root root     875 Jul 25 21:19 brcmfmac43430-sdio.AP6212.txt
lrwxrwxrwx 1 root root      31 Jul 25 21:19 brcmfmac43430-sdio.bin -> ../cypress/cyfmac43430-sdio.bin
lrwxrwxrwx 1 root root      36 Jul 25 21:19 brcmfmac43430-sdio.clm_blob -> ../cypress/cyfmac43430-sdio.clm_blob
lrwxrwxrwx 1 root root      29 Jan  2 08:20 brcmfmac43430-sdio.friendlyarm,nanopi-r1.txt -> brcmfmac43430-sdio.AP6212.txt
...
```

Comment (chombourger) [reply meta-isar/recipes-bsp/firmware-nanopi-r1/firmware-nanopi-r1.bb:13]: and
```
root@mtda:/# dpkg -S /lib/firmware/brcm/brcmfmac43430-sdio.friendlyarm,nanopi-r1.txt 
firmware-nanopi-r1: /lib/firmware/brcm/brcmfmac43430-sdio.friendlyarm,nanopi-r1.txt
```

**Comment by chombourger** [comment]
We should amend the sumary line of both commits: `Add` -> `add`
and `WiFi` -> `Wi-Fi`

```
f125ebd nanopi-r1: add support for Bluetooth
132283d nanopi-r1: add support for Wi-Fi
```

File: meta-isar/recipes-bsp/firmware-nanopi-r1/firmware-nanopi-r1.bb (hunk 4 above)
Comment (vj-kumar) [reply meta-isar/recipes-bsp/firmware-nanopi-r1/firmware-nanopi-r1.bb:13]: Sure. This also works.

**Comment by vj-kumar** [comment]
> We should amend the sumary line of both commits: `Add` -> `add` and `WiFi` -> `Wi-Fi`
> 
> ```
> f125ebd nanopi-r1: add support for Bluetooth
> 132283d nanopi-r1: add support for Wi-Fi
> ```

Done.

**Review by chombourger — APPROVED** [review]

**Pull request merged**